sqs_workers/memory_sqs.py

```diff
@@ -158,19 +160,40 @@ def delete_messages(self, Entries):
         message_ids = {entry["Id"] for entry in Entries}
 
         successfully_deleted = set()
-        push_back_messages = []
 
-        for message in self.messages:
+        for i, message in enumerate(self.in_flight):
             if message.message_id in message_ids:
                 successfully_deleted.add(message.message_id)
-            else:
-                push_back_messages.append(message)
-        self.messages[:] = push_back_messages
+                del self.in_flight[i]
 
-        didnt_deleted = message_ids.difference(successfully_deleted)
         return {
             "Successful": [{"Id": _id} for _id in successfully_deleted],
-            "Failed": [{"Id": _id} for _id in didnt_deleted],
+        }
+
+    def change_message_visibility_batch(self, Entries):
+        """
+        Changes message visibility by looking at in-flight messages, setting
+        a new execute_at, and returning it to the pool of processable messages
+        """
+        edited = []
+        return_to_pool = []
+        entries_by_id = {e["Id"]: e for e in Entries}
+
+        for i, m in enumerate(self.in_flight):
+            if m.message_id in entries_by_id.keys():
+                sec = int(entries_by_id[m.message_id]["VisibilityTimeout"])
+                now = datetime.datetime.utcnow()
+                execute_at = now + datetime.timedelta(seconds=sec)
+                changed = attr.evolve(m, execute_at=execute_at)
```

Comment:
Nit and subjective, do feel free to ignore this one, but considering we use the `edited` verbiage below, I found this name confusing (one's a message, the other's all messages) and wondered if `message`  wouldn't be less distracting.

Reply:
Yeah indeed! I kinda threw this together (my understanding is that `MemoryQueue` is only used for tests, it's quite half-baked) but I am still happy to make changes to have it be more clear